tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -951,6 +959,49 @@ void test_tls_pointer(void)
 	}
 }
 
+void user_access_queue_private_data(void *p1, void *p2, void *p3)
```

Comment:
can we put this in the k_queue tests and not in this test case?

I know there's some pipe code already here that needs to be moved as well

Reply:
Do you mean move to tests/kernel/queue/?

Reply:
Done, in that PR added test to the kernel/queue directory.

Reply:
@andrewboie I moved kernel/queue test to the separate PR #29188